versioned_docs/version-5.1/general-concepts/multilingual.md

```diff
@@ -1,3 +1,202 @@
 Multilingual
 =======================
-This is the content for multilingual
\ No newline at end of file
+
+Joomla provides robust support for multiple languages, allowing developers to create extensions that can be easily translated and used in different languages.
+
+## How Joomla Supports Multiple Languages
+
+Joomla differentiates between **installed languages** (languages that the Joomla core or extensions are translated into) and **content languages** (languages that site content can be written in). 
+
+This distinction helps in managing translations more effectively.
+
+### Language Constants
+
+To handle translations, Joomla uses **language constants**. These constants are defined in language files (`.ini` and `.sys.ini`) and are used throughout the codebase. 
+
+By using **constants**, you can ensure that your extensions are easily translatable.
+
+### Language Files
+
+**Language files** are simple text files that map language constants to their translations. There are two main types of language files:
+- `.ini` files: Used for **regular translations.**
+- `.sys.ini` files: Used for **installation and system messages.**
+
+Here’s an example of a language file (`mod_example.ini`):
+```ini
+MOD_EXAMPLE_HELLO="Hello"
+MOD_EXAMPLE_GOODBYE="Goodbye"
+```
+
+Place it in the `language/en-GB` directory of your extension.
+
+
+## Implementing Multilingual Support in Your Extension
+
+### Where Language Files are stored
+
+Language files can be stored in the following directories:
+- **Components:** `language/en-GB/com_example.ini`
+- **Modules:** `language/en-GB/mod_example.ini`
+- **Plugins:** `language/en-GB/plg_example.ini`
+- **Templates:** `language/en-GB/tpl_example.ini`
+- **Language Root**: `/language` or `/administrator/language`
+
```

Comment:
This needs clarification
 
It says Stored in directories but lists files

You should also add information about when you should create the file in /language or /administrator/language